src/ocaml_lsp.ml

```diff
@@ -53,64 +53,76 @@ type t =
 
 let get_version_from_serverInfo { serverInfo; experimental_capabilities = _ } =
   match serverInfo with
-  | None -> Error `Missing_serverInfo (* ocamllsp should have [serverInfo] *)
+  | None -> None
   | Some { name; version } ->
-    if String.equal name "ocamllsp" then
-      match version with
-      | None -> Error `ServerInfo_version_missing
-      | Some v -> Ok v
+    if String.equal name "ocamllsp" then version
     else (
       log_chan ~section:"Ocaml_lsp.get_version" `Warn
         "the language server is not ocamllsp";
       (* practically impossible but let's be defensive *)
-      Error `Language_server_isn't_ocamllsp)
+      None)
 
-let get_version_semver t =
-  match get_version_from_serverInfo t with
-  | Ok v -> (
-    match String.split v ~on:'-' |> List.hd with
-    | Some v -> Ok v
-    | None -> Error `Unable_to_parse_version)
-  | Error _ as err -> err
+let lsp_versions =
+  let main =
+    [ ( (4, 12)
+      , [| "1.5.0"
+         ; "1.6.0"
+         ; "1.6.1"
+         ; "1.7.0"
+         ; "1.8.0"
+         ; "1.8.1"
+         ; "1.8.2"
+         ; "1.8.3"
+         ; "1.9.0"
+        |] )
+    ; ((4, 13), [| "1.9.1"; "1.10.0"; "1.10.1" |])
```

Comment:
Btw, ocamlc version library should have values corresponding to ocamlc versions, ie we could use `Ocaml_version.v4_12` instead of `(4, 12)`

Reply:
>  ie we could use Ocaml_version.v4_12 instead of (4, 12)

This could work, but is it really clarifying anything? We are mapping version OCaml versions prefixes to ocamllsp versions.

I'll add some comments